lib/TH/THMemoryFile.c

```diff
@@ -177,7 +177,7 @@ static int THMemoryFile_mode(const char *mode, int *isReadable, int *isWritable)
       size_t i;                                                           \
       for(i = 0; i < n; i++)                                            \
       {                                                                 \
-        ssize_t nByteWritten;                                           \
+        int nByteWritten;                                               \
```

Comment:
For the `Byte` and `Char` types, `nByteWritten` can be affected the value of `n` which is of type `size_t`. Can't this overflow an `int`?